tests/Unit/IO/Test_StellarCollapseEos.cpp

```diff
@@ -0,0 +1,376 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <boost/iterator/iterator_facade.hpp>
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include "DataStructures/BoostMultiArray.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/Header.hpp"  // IWYU pragma: keep
+#include "IO/H5/StellarCollapseEos.hpp"
+#include "Informer/InfoFromBuild.hpp"
+#include "tests/Unit/TestingFramework.hpp"
+
+namespace {
+
+// For 1D arrays the numbers (in, for example, logrho_data) are the first and
+// second numbers in the corresponding dataset. For 3D arrays they are the first
+// 2x2x2 block (so all elements with indices [0][0][0] -> [1][1][1])
+void test_tabulated(const std::string& file_path,
+                    const std::string& subgroup_path) noexcept {
+  h5::H5File<h5::AccessType::ReadOnly> sample_file(unit_test_path() +
+                                                   file_path);
+  const auto& sample_data =
+      sample_file.get<h5::StellarCollapseEos>(subgroup_path);
+
+  const size_t energy_shift_data = 317;
+  const auto energy_shift_from_file =
+      sample_data.get_scalar_dataset("energy_shift");
+  CHECK(energy_shift_from_file == energy_shift_data);
+
+  const size_t have_rel_cs2_data = 1;
+  const auto have_rel_cs2_from_file =
+      sample_data.get_scalar_dataset("have_rel_cs2");
+  CHECK(have_rel_cs2_from_file == have_rel_cs2_data);
+
+  const size_t pointsrho_data = 391;
+  const auto pointsrho_from_file = sample_data.get_scalar_dataset("pointsrho");
+  CHECK(pointsrho_from_file == pointsrho_data);
+
+  const size_t pointstemp_data = 163;
+  const auto pointstemp_from_file =
+      sample_data.get_scalar_dataset("pointstemp");
+  CHECK(pointstemp_from_file == pointstemp_data);
+
+  const size_t pointsye_data = 66;
+  const auto pointsye_from_file = sample_data.get_scalar_dataset("pointsye");
+  CHECK(pointsye_from_file == pointsye_data);
+
+  const std::vector<double> logrho_data = {3.0239960056064277,
+                                           3.0573293389397609};
+  const auto logrho_from_file = sample_data.get_rank1_dataset("logrho");
+  CHECK(logrho_from_file == logrho_data);
+
+  const std::vector<double> logtemp_data = {-3.0, -2.9666666666666668};
+  const auto logtemp_from_file = sample_data.get_rank1_dataset("logtemp");
+  CHECK(logtemp_from_file == logtemp_data);
+
+  const std::vector<double> ye_data = {0.005, 0.015};
+  const auto ye_from_file = sample_data.get_rank1_dataset("ye");
+  CHECK(ye_from_file == ye_data);
+
+  boost::multi_array<double, 3> Abar_data(boost::extents[2][2][2]);
+  const auto Abar_from_file = sample_data.get_rank3_dataset("Abar");
+  CHECK(std::all_of(Abar_from_file.data(),
```

Comment:
prefer `alg::all_of` in `Utilities/Algorithm.hpp`. Also, don't pass pointers to `all_of`, pass the begin and end iterators instead.